src/SprykerEco/Zed/Unzer/Business/Payment/Resolver/UnzerMarketplacePaymentUnzerCredentialsResolver.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Business\Payment\Resolver;
+
+use Generated\Shared\Transfer\QuoteTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsConditionsTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsCriteriaTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsTransfer;
+use Generated\Shared\Transfer\UnzerMarketplacePaymentCredentialsResolverCriteriaTransfer;
+use SprykerEco\Shared\Unzer\UnzerConstants;
+use SprykerEco\Zed\Unzer\Business\ApiAdapter\UnzerPaymentMethodsAdapterInterface;
+use SprykerEco\Zed\Unzer\Business\Exception\UnzerException;
+use SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface;
+
+class UnzerMarketplacePaymentUnzerCredentialsResolver implements UnzerMarketplacePaymentUnzerCredentialsResolverInterface
+{
+ /**
+  * @var \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface
+  */
+    protected UnzerReaderInterface $unzerReader;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\ApiAdapter\UnzerPaymentMethodsAdapterInterface
+     */
+    protected UnzerPaymentMethodsAdapterInterface $unzerPaymentMethodsAdapter;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface $unzerReader
+     * @param \SprykerEco\Zed\Unzer\Business\ApiAdapter\UnzerPaymentMethodsAdapterInterface $unzerPaymentMethodsAdapter
+     */
+    public function __construct(UnzerReaderInterface $unzerReader, UnzerPaymentMethodsAdapterInterface $unzerPaymentMethodsAdapter)
+    {
+        $this->unzerReader = $unzerReader;
+        $this->unzerPaymentMethodsAdapter = $unzerPaymentMethodsAdapter;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UnzerMarketplacePaymentCredentialsResolverCriteriaTransfer $unzerMarketplacePaymentCredentialsResolverCriteriaTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerCredentialsTransfer
+     */
+    public function resolveMarketplacePaymentUnzerCredentials(
```

Comment:
this method is now used at expander
it throws an exception and affects already existing facade method `performPreSaveOrderStack` at least
this change is BC-breaking since this facade method never threw this exception before

Reply:
Fixed.